File: Sem3/PY/Lista11/zad1.py
import mysql.connector
from secrets import host, user, password

db = mysql.connector.connect(user=user, password=password, host=host, database="pythoncourse")
cursor = db.cursor()

# cursor.execute("CREATE DATABASE pythoncourse")

# cursor.execute("SHOW DATABASES")
# for x in cursor:
#   print(x)


cursor.execute('''
CREATE TABLE IF NOT EXISTS books (
    id INT AUTO_INCREMENT PRIMARY KEY,
    author VARCHAR(255) NOT NULL,
    title VARCHAR(255) NOT NULL,
    year INT NOT NULL
)
''')

cursor.execute('''
CREATE TABLE IF NOT EXISTS friends (
    id INT AUTO_INCREMENT PRIMARY KEY,
    name VARCHAR(255) NOT NULL,
    email VARCHAR(255) NOT NULL
)
''')

cursor.execute('''
CREATE TABLE IF NOT EXISTS borrowings (
    id INT AUTO_INCREMENT PRIMARY KEY,
    book_id INT NOT NULL,
    friend_id INT NOT NULL,
               
    FOREIGN KEY (book_id) REFERENCES books(id),
    FOREIGN KEY (friend_id) REFERENCES friends(id)
)
''')

def add_book(author, title, year):
    query = "INSERT INTO books (author, title, year) VALUES (%s, %s, %s)"
    values = (author, title, year)
    cursor.execute(query, values)
    db.commit()
    print("Book added successfully")

def borrow_book(book_id, friend_id):
    query = "INSERT INTO borrowings (book_id, friend_id) VALUES (%s, %s)"
    values = (book_id, friend_id)
    cursor.execute(query, values)
    db.commit()
    print("Book borrowed successfully")

def return_book(book_id, friend_id):
    query = "DELETE FROM borrowings WHERE book_id = %s AND friend_id = %s"
    values = (book_id, friend_id)
    cursor.execute(query, values)
    db.commit()
    print("Book returned successfully")

File: Sem3/WEPPO/Lista9/zad1.js
let mysql = require('mysql');

let con = mysql.createConnection({
    host: "localhost",
    user: "root",
    password: "admin",
    database: "weppo",
});

function drop_table() {
    let sql = "DROP TABLE IF EXISTS PEARSON;";
    con.query(sql, function(err, result) {
        if (err) throw err;
        console.log("Table dropped");
    });
    let sql2 = "DROP TABLE IF EXISTS PEARSON2;";
    con.query(sql2, function(err, result) {
        if (err) throw err;
        console.log("Table dropped");
    });
}

function create_table() {
    let sql = "CREATE TABLE PEARSON (id INT AUTO_INCREMENT PRIMARY KEY, name VARCHAR(255), surname VARCHAR(255), age INT, sex VARCHAR(1));";
    con.query(sql, function(err, result) {
        if (err) throw err;
        console.log("Table created");
    });

    /* //! To create sequence in PostgreSQL, which does not exist in MySQL since it is using AUTO_INCREMENT:
        CREATE SEQUENCE seq_person_id
        START WITH 1
        INCREMENT BY 1;

        INSERT INTO PEARSON(id, name, surname, age, sex)
        VALUES (NEXTVAL('seq_person_id'), 'John', 'Doe', 30, 'M');
    */

    // simulation of sequence in MySQL:
    let sql2 = "CREATE TABLE PEARSON2 (id INT PRIMARY KEY, name VARCHAR(255), surname VARCHAR(255), age INT, sex VARCHAR(1));";
    con.query(sql2, function(err, result) {
        if (err) throw err;
        console.log("Table2 created");
    });
}

function add_person(name, surname, age, sex) {
    let sql = `INSERT INTO PEARSON (name, surname, age, sex) VALUES ('${name}', '${surname}', '${age}', '${sex}')`;
    con.query(sql, function(err, result) {
        if (err) throw err;
        console.log("Person added");
    });

    // simulation of sequence in MySQL:
    let sql2 = `INSERT INTO PEARSON2(id, name, surname, age, sex) SELECT IFNULL((SELECT MAX(id) FROM PEARSON), 0) + 1, '${name}', '${surname}', '${age}', '${sex}';`;
    con.query(sql2, function(err, result) {
        if (err) throw err;
        console.log("Person added");
    });
}

con.connect(function(err) {
    if (err) throw err;
    console.log("Connected!");

    drop_table();
    create_table();

    const pearsons = [
        ["John", "Doe", 30, "M"],
        ["Jane", "Doe", 25, "F"],
        ["John", "Smith", 40, "M"],
        ["Jane", "Smith", 35, "F"],
    ];

    pearsons.forEach(pearson => {
        add_person(pearson[0], pearson[1], pearson[2], pearson[3]);
    });

    con.query("SELECT * FROM PEARSON", function(err, result, fields) {
        if (err) throw err;
        console.log("------------ SELECT * FROM PEARSON ------------");
        console.log(result);
    });

    con.query("SELECT * FROM PEARSON WHERE surname = 'Doe'", function(err, result, fields) {
        if (err) throw err;
        console.log("------------ SELECT * FROM PEARSON WHERE surname = 'Doe' ------------");
        console.log(result);
    });
});
